system/src/commonMain/kotlin/com/splendo/kaluga/system/network/NetworkStateRepo.kt

```diff
@@ -0,0 +1,58 @@
+/*
+ Copyright 2020 Splendo Consulting B.V. The Netherlands
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+ */
+
+package com.splendo.kaluga.system.network
+
+import co.touchlab.stately.concurrency.AtomicReference
+import com.splendo.kaluga.state.ColdStateRepo
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+import kotlin.coroutines.CoroutineContext
+
+class NetworkStateRepo(
+    context: Any?,
+    coroutineContext: CoroutineContext = Dispatchers.Main
+) : ColdStateRepo<NetworkState>(coroutineContext = coroutineContext) {
+
+    private var _lastKnownNetwork = AtomicReference<Network>(Network.Absent)
```

Comment:
Not a big fan of underscore prefixes, also it says nothing about difference between private and internal variables. What is Kotlin best practice for backed fields here?

Reply:
https://kotlinlang.org/docs/reference/coding-conventions.html#function-names

Reply:
👍

Reply:
I'm using the underscore prefix because the difference between this and the property below is that the first one is wrapped in `AtomicReference`. The second one is used to not directly access the AtomicReference property and so get and set it without using `_lastKnownNetwork.get/set`.

Reply:
We indeed use this in several places in kaluga, for atomic references but also as a pattern in viewmodels.